src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -27,35 +28,45 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
             throw new InvalidArgumentError();
         }
 
-        return this._paypalScriptLoader.loadPaypalCommerce(paypalOptions.clientId)
-            .then(paypal => {
-                return paypal.Buttons({
-                    createOrder: () => this._setupPayment(),
-                    onApprove: data => this._tokenizePayment(paymentMethod.id, data),
-                }).render(`#${options.containerId}`);
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(state => {
+                const config = state.config.getStoreConfig();
+                const cart = state.cart.getCart();
+
+                if (!config) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                }
+
+                const paramsScript = {
+                    'client-id': paypalOptions.clientId,
+                    currency: config.currency.code,
+                };
+
+                return this._paypalScriptLoader.loadPaypalCommerce(paramsScript)
+                    .then(paypal => {
+                        return paypal.Buttons({
+                            createOrder: () => this._setupPayment(cart),
+                            onApprove: data => this._tokenizePayment(paymentMethod.id, data),
+                        }).render(`#${options.containerId}`);
+                    });
             });
     }
 
     deinitialize(): Promise<void> {
         return Promise.resolve();
     }
 
-    private _setupPayment(): Promise<string> {
-
-        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
-            .then(state => {
-                const url = '/api/storefront/payment/paypalcommerce';
-                const headers = {
-                    'X-API-INTERNAL': INTERNAL_USE_ONLY,
-                    'Content-Type': 'application/x-www-form-urlencoded',
-                };
-                const cart = state.cart.getCart();
-                const currentCartId = cart ? cart.id : '';
+    private _setupPayment(cart?: Cart): Promise<string> {
```

Comment:
Can `cart` ever be optional? If it's not defined, it means the data hasn't be loaded for some reason and we should throw an error to stop the code execution (similar to what you're doing with the missing config check).

Reply:
You're right. I added check to cart
Thank you